src/panels/lovelace/components/hui-conditional.ts

```diff
@@ -0,0 +1,154 @@
+import { LitElement } from "lit-element";
+import { HomeAssistant } from "../../../types";
+import { LovelaceCard } from "../types";
+import { LovelaceCardConfig, Condition } from "../../../data/lovelace";
+import { LovelaceElement, LovelaceElementConfig } from "../elements/types";
+import { EntityRowConfig } from "../entity-rows/types";
+import { createCardElement } from "../common/create-card-element";
+// import { createRowElement } from "../common/create-row-element";
+
+export interface ConditionalConfig {
+  conditions: Condition[];
+  card: LovelaceCardConfig;
+  elements: LovelaceElementConfig[];
+  rows: EntityRowConfig[];
+  // todo: add row (can't find any createhuirow or LovelaceRowConfig...)
+  // rows: Lovelace;
+  // use it with createRowElement
+}
+
+export class HuiConditional extends LitElement {
+  protected _card?: LovelaceCard;
+
+  private _hass?: HomeAssistant;
+  private _conditionalConfig?: ConditionalConfig;
+  private _elements: LovelaceElement[] = [];
+
+  set hass(hass: HomeAssistant) {
+    this._hass = hass;
+
+    if (this.hasNoItems()) {
+      return;
+    }
+
+    const visible = this.handleItems();
+
+    // This will hide the complete component so it won't get styled by parent in case there's nothing in it
+    this.style.setProperty("display", visible ? "" : "none");
+  }
+
+  public setConditionalConfig(config) {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.conditions.every((c) => c.entity && (c.state || c.state_not)) ||
+      (!config.card && (!config.elements || !Array.isArray(config.elements)))
+    ) {
+      throw new Error("Error in conditional configuration.");
+    }
+
+    this._conditionalConfig = config;
+
+    // set style (overridable)
+    // LitElement static styles can't work since we want to allow deriving classes to modify it
```

Comment:
Not true, static styles can just add super.styles to their CSS array. Make sure the return set is set to `CSSResultArray`

Reply:
Actually, why would we even need this to begin with ? The deriving classes need to handle CSS, this class should not be aware. 

Also, a conditional card really should not have ANY CSS. It's only showing something else.